pkg/datadogclient/client.go

```diff
@@ -36,6 +36,43 @@ func InitDatadogMonitorClient(logger logr.Logger, creds config.Creds) (DatadogMo
 		return DatadogMonitorClient{}, errors.New("error obtaining API key and/or app key")
 	}
 
+	configV1 := datadogapi.NewConfiguration()
+	apiClient := datadogapi.NewAPIClient(configV1)
+	client := datadogV1.NewMonitorsApi(apiClient)
+
+	authV1, err := setupAuth(logger, creds)
+	if err != nil {
+		return DatadogMonitorClient{}, err
+	}
+
+	return DatadogMonitorClient{Client: client, Auth: authV1}, nil
+}
+
+// DatadogSLOClient contains the Datadog Monitor API Client and Authentication context.
+type DatadogSLOClient struct {
+	Client *datadogV1.ServiceLevelObjectivesApi
+	Auth   context.Context
+}
+
+// InitDatadogSLOClient initializes the Datadog SLO API Client and establishes credentials.
+func InitDatadogSLOClient(logger logr.Logger, creds config.Creds) (DatadogSLOClient, error) {
+	if creds.APIKey == "" || creds.AppKey == "" {
+		return DatadogSLOClient{}, errors.New("error obtaining API key and/or app key")
+	}
+
+	configV1 := datadogapi.NewConfiguration()
+	apiClient := datadogapi.NewAPIClient(configV1)
+	client := datadogV1.NewServiceLevelObjectivesApi(apiClient)
```

Comment:
What's the `auth` for out of curiosity?
Could we not use a generic client (for metrics, monitors and SLOs?)

Reply:
we talked IRL, but changes in the new client version have resulted in separate clients for these APIs